client/client-lib/src/mint/mod.rs

```diff
@@ -101,6 +105,28 @@ impl<'c> MintClient<'c> {
             .collect()
     }
 
+    fn new_note_secret(&self, dbtx: &mut DatabaseTransaction<'_>) -> DerivableSecret {
```

Comment:
We should have a test that runs this in parallel for many threads in loop and verify that it's atomic.

What's consequence of re-using same nonce twice? Any bad consequences of skipping a nonce?

Reply:
Good idea!

Consequences:
* Reuse: burning money
* Skipping: need to account for that when recovering (gap limit)

Reply:
#861 created, as I just landed this PR

Reply:
Really good instinct! Had to fix a bug with rocksdb isolation c8597b644451cfc1c1955c7b0455784c6847ab3a. @m1sterc001guy did we just miss this or should we not need snapshot isolation and is that even the right term for it?